pkg/controller/apischeme/apischeme_controller.go

```diff
@@ -96,5 +119,320 @@ func (r *ReconcileAPIScheme) Reconcile(request reconcile.Request) (reconcile.Res
 		// Error reading the object - requeue the request.
 		return reconcile.Result{}, err
 	}
-	return reconcile.Result{}, err
+	// If the management API isn't enabled, we have nothing to do!
+	// TODO (lisa/lseelye): This should call a teardown feature to ensure we have
+	// disabled, but that has SERIOUS potential issues with Hive, as it will come to
+	// depend on rh-api.
+	if !instance.Spec.ManagementAPIServerIngress.Enabled {
+		return reconcile.Result{}, nil
+	}
+
+	ownerTags, err := utils.AWSOwnerTag(r.client)
+	if err != nil {
+		reqLogger.Error(err, "Couldn't get the cluster owner tags")
+		SetAPISchemeStatus(reqLogger, instance, "Couldn't reconcile", "Couldn't get the cluster owner tags", cloudingressv1alpha1.ConditionError)
+		r.client.Status().Update(context.TODO(), instance)
+		return reconcile.Result{}, err
+	}
+
+	region, err := utils.GetClusterRegion(r.client)
+	if err != nil {
+		reqLogger.Error(err, "Couldn't get the cluster owner tags")
+		SetAPISchemeStatus(reqLogger, instance, "Couldn't reconcile", "Couldn't get the cluster's AWS region", cloudingressv1alpha1.ConditionError)
+		r.client.Status().Update(context.TODO(), instance)
+		return reconcile.Result{}, err
+	}
+	// We expect this secret to exist in the same namespace Account CR's are created
+	// TODO: Get the region of the cluster
+	awsClient, err := awsclient.GetAWSClient(r.client, awsclient.NewAwsClientInput{
+		SecretName: config.AWSSecretName,
+		NameSpace:  config.OperatorNamespace,
+		AwsRegion:  region,
+	})
+	if err != nil {
+		reqLogger.Error(err, "Failed to get AWS client")
+		SetAPISchemeStatus(reqLogger, instance, "Couldn't reconcile", "Couldn't create an AWS client", cloudingressv1alpha1.ConditionError)
+		r.client.Status().Update(context.TODO(), instance)
+		return reconcile.Result{}, err
+	}
+	subnets, err := utils.GetMasterNodeSubnets(r.client)
+	if err != nil {
+		reqLogger.Error(err, "Couldn't get the subnets used by master nodes")
+		SetAPISchemeStatus(reqLogger, instance, "Couldn't reconcile", "Couldn't get the cluster's subnets", cloudingressv1alpha1.ConditionError)
+		r.client.Status().Update(context.TODO(), instance)
+		return reconcile.Result{}, err
+	}
+
+	clusterBaseDomain, err := utils.GetClusterBaseDomain(r.client)
```

Comment:
you can also get baseDomain from the DNS object

Reply:
I tried and it doesn't seem to work. :(